client/webserver/site/src/js/markets.ts

```diff
@@ -2027,51 +2077,175 @@ export default class MarketsPage extends BasePage {
     this.balanceWgt.updateAsset(this.openAsset.id)
   }
 
-  /* lotChanged is attached to the keyup and change events of the lots input. */
-  lotChanged () {
+  lotFieldKeyUpEvent () {
     const page = this.page
-    const lots = parseInt(page.lotField.value || '0')
-    if (lots <= 0) {
-      page.lotField.value = '0'
+
+    // We can't do anything with empty value here, and since it is a default
+    // we don't want to report is to be an issue either. Just make sure to
+    // reset both lots and quantity because both represent the same thing.
+    if (page.lotField.value === '') {
       page.qtyField.value = ''
-      this.previewQuoteAmt(false)
       return
     }
-    const lotSize = this.market.cfg.lotsize
-    page.lotField.value = String(lots)
-    // Conversion factor must be a multiple of 10.
-    page.qtyField.value = String(lots * lotSize / this.market.baseUnitInfo.conventional.conversionFactor)
-    this.previewQuoteAmt(true)
+
+    const [inputInvalid, adjusted,, adjQty] = this.parseLotInput()
+    if (inputInvalid || adjusted) {
+      // Let the user know that lot value he's entered was rounded down to the
+      // nearest integer number.
+      highlightOutlineRed(page.lotField)
+      highlightBackgroundRed(page.lostSizeBox)
+    }
+    if (inputInvalid) {
+      page.orderTotalPreview.textContent = ''
+      page.qtyField.value = ''
+      return
+    }
+    // Lots and quantity fields are tightly coupled to each other, when one is
+    // changed, we need to update the other one as well. Lot has already been
+    // updated by the user, update the other one here.
+    page.qtyField.value = String(adjQty)
+
+    this.previewTotal()
   }
 
-  /*
-   * quantityChanged is attached to the keyup and change events of the quantity
-   * input.
-   */
-  quantityChanged (finalize: boolean) {
+  lotFieldChangeEvent () {
```

Comment:
The input `--5` in the lots input does not show an error indication and does not self-adjust to zero.

On master, a negative sign is removed immediately after entering. That feels better.

Reply:
Should work better now,

but I still have quite a bit of minor things to address to make sure different  edge-cases are covered and overall experience is smooth.